compiler-rt/lib/builtins/CMakeLists.txt

```
@@ -854,10 +854,12 @@ else ()
       cmake_push_check_state()
       # TODO: we should probably make most of the checks in builtin-config depend on the target flags.
       set(BUILTIN_CFLAGS_${arch} ${BUILTIN_CFLAGS})
-      # CMAKE_REQUIRED_FLAGS must be a space separated string but unlike TARGET_${arch}_CFLAGS,
-      # BUILTIN_CFLAGS_${arch} is a CMake list, so we have to join it to create a valid command line.
-      list(JOIN BUILTIN_CFLAGS " " CMAKE_REQUIRED_FLAGS)
-      set(CMAKE_REQUIRED_FLAGS "${TARGET_${arch}_CFLAGS} ${BUILTIN_CFLAGS_${arch}}")
+      # CMAKE_REQUIRED_FLAGS must be a space separated string
+      # Join BUILTIN_CFLAGS_${arch} and TARGET_${arch}_CFLAGS as a
+      # space-separated string.
+      set(CMAKE_REQUIRED_FLAGS ${BUILTIN_CFLAGS_${arch}})
+      list(APPEND CMAKE_REQUIRED_FLAGS ${TARGET_${arch}_CFLAGS})
+      list(JOIN CMAKE_REQUIRED_FLAGS " " CMAKE_REQUIRED_FLAGS)
```

How about the following slightly simpler version?
```suggestion
      # CMAKE_REQUIRED_FLAGS must be a space separated string
      # Join BUILTIN_CFLAGS_${arch} and TARGET_${arch}_CFLAGS as a
      # space-separated string.
      set(_required_flags_list ${TARGET_${arch}_CFLAGS} ${BUILTIN_CFLAGS_${arch}})
      list(JOIN _required_flags_list " " CMAKE_REQUIRED_FLAGS)
```